torch/_dynamo/variables/tensor.py

```diff
@@ -154,10 +154,37 @@ def var_getattr(self, tx, name):
             result = self.call_method(tx, "dim", [], {})
         elif name == "data":
             result = self.call_method(tx, "detach", [], {})
-        elif name == "T":
-            args = [variables.ConstantVariable(i) for i in range(self.ndim - 1, -1, -1)]
-            result = self.call_method(tx, "permute", args, {})
-
+        elif name in ("T", "H"):
+            out = (
+                tx.output.create_proxy(
+                    "call_method",
+                    "conj",
+                    *proxy_args_kwargs([self], {}),
+                )
+                if name == "H"
+                else self
+            )
+            args_list = [
+                variables.ConstantVariable(i) for i in range(self.ndim - 1, -1, -1)
+            ]
+            args = [variables.TupleVariable(args_list)]
+            result = out.call_method(tx, "permute", args, {})
+        elif name in ("mT", "mH"):
+            out = (
+                tx.output.create_proxy(
+                    "call_method",
+                    "conj",
+                    *proxy_args_kwargs([self], {}),
+                )
+                if name == "mH"
+                else self
+            )
+            dims = (-2, -1) if self.ndim > 0 else (-1, 0)
```

Comment:
Wait, is this correct? Shouldn't we just return `out` in this case. When I call `permute` on a 0D tensor I get a particularly weird error:
```python
>>> torch.randn(()).permute(-1, 0)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: permute(sparse_coo): number of dimensions in the tensor input does not match the length of the desired ordering of dimensions i.e. input.dim() = 0 is not equal to len(dims) = 2
```

Reply:
huh this was working before, but yeah best to just return out agreed